.gitignore

```diff
@@ -23,7 +23,7 @@ release.h
 src/transfer.sh
 src/configs
 redis.ds
-src/redis.conf
+src/placeholderkv.conf
```

Comment:
I had a discussion with @zuiderkwast. we should create symlinks for these files for two reasons 

1. to reduce the size of the pr and we can change the names in the rest of the codebase gradually
2. to maintain compat. I think the "redis" string is spread out in the whole ecosystem already.

Reply:
IIUC, you're suggesting to have a follow up task to create a symlink to redis-* binaries with valkey-* binaries and the config file. Is that right?

Reply:
For the configuration file there is no makefile-based install currently (so nowhere really to create this new symlink) - on Linux this is handled by the distributions in their rpm spec / debian rules files.  On rpm systems (at least, maybe deb also) its slightly more complex in that there's also a directory created for the config files - i.e. redis.conf and sentinel.conf are placed in /etc/redis (so the directory will need to be symlinked too).
For valkey I recommend leaving this part of the symlinking to the distributors, but I agree it'll be good to install compatibility symlinks for the commands.

Reply:
Thanks for the insight into Linux @natoscott

I also did think about putting the symlink creation as part of Makefile build. 

But then I thought there could also be users who directly download the release from github and start working on it and redis.conf would be missing until they perform a build. So, proceeded with this approach.

Reply:
@natoscott, I like your idea of leaving the symlinks to the distributors but curious to know why you don't think creating symlinks in the makefile would work?

I think there are 3 ways for a user to get access to the valkey binaries

1. pre-installed 
2. pre-built docker container
3. github repo and DIY

I would assume that all end users fall in the first 2 categories while the 3rd bucket is probably all developers. the compat bars will be different for these two personas, end users vs developers, with the bar a lot higher for the end user. So maybe having the distributors (including docker container distributors) own up the symlinks is a better idea? developers should be able to figure out where the files are quickly.